Translate doct/testing/debugging

oozoofrog opened the pull request:
#31 
- [x] Introduction
- [x] The Dart Analyzer
- [x] Dart Observatory (statement-level single-stepping debugger and profiler)
  - [x] debugger() statement
- [ ] print and debugPrint with flutter logs
- [ ] Debug mode assertions
- [ ] Debugging application layers
  - [ ] Widget layer
  - [ ] Rendering layer
  - [ ] Layers
  - [ ] Semantics
  - [ ] Scheduling
- [ ] Visual debugging
- [ ] Debugging animations
- [ ] Debugging performance problems
  - [ ] Measuring app startup time
  - [ ] Tracing any Dart code performance
- [ ] PerformanceOverlay
- [ ] Material grid
- [ ] Common problems
  - [ ] “Too many open files” exception (MacOS)

Commit:
Translate Introduction

Commit:
- [x] The Dart Analyzer

bro-jo commented on src/docs/testing/debugging.md:
```diff
@@ -1,28 +1,21 @@
+
+The Dart Analyzer는 문제를 추적하기 위해 여러분이 코드에 넣어둔 타입 어노테이션을 
+이용합니다. 문제를 추적하기 위한 가장 빠르고 덜 고통스러운 방법이니만큼 모든 곳에서
+var 선언,  타입 없는 인수, 타입 없는 배열 문법 등의 사용을 피하는 것을 권장합니다.
```
```
var 선언-> `var` 선언
```

oozoofrog replied:
더 읽기 좋네요. 고맙습니다

bro-jo commented on src/docs/testing/debugging.md:
```diff
@@ -1,28 +1,21 @@
+
+The Dart Analyzer는 문제를 추적하기 위해 여러분이 코드에 넣어둔 타입 어노테이션을 
+이용합니다. 문제를 추적하기 위한 가장 빠르고 덜 고통스러운 방법이니만큼 모든 곳에서
+var 선언,  타입 없는 인수, 타입 없는 배열 문법 등의 사용을 피하는 것을 권장합니다.
```
이 부분이 뭔가 저한테는 어색하게 느껴지네요.
```suggestion
var 선언,  타입 없는 인수, 타입 없는 배열 문법 등의 사용을 자제하는 것이 좋습니다.
```

Commit:
리뷰를 반영했습니다.

Commit:
- [x] Dart Observatory (statement-level single-stepping debugger and profiler)
  - [ ]  `debugger()` statement

번역 상태가 그렇게 좋지 않네요. 마지막 문장이 특히 어려운데 이건 제가 플러터의 디버그 프로파일링을 사용해보고 나면 개선할 수 있을거 같습니다. 미리 알려주시면 더 좋구요 ㅎㅎ

Commit:
- [x] Dart Observatory (statement-level single-stepping debugger and profiler)
  - [x]  `debugger()` statement

bro-jo commented:
한 분 더 리뷰해주시면 머지하겠습니다

wanbok reviewed (approved):
LGTM
